### PR TITLE
Help cmd params

### PR DESCRIPTION
**Pull request by richsalz**
This adds documentation for all commands that take parameters.

It makes the help strings consistent.

It is based on #9953 so only look at the top commit.  Or, better yet, go review and approve that one first.

### CODE REVIEW

**Comment by richsalz** [comment]
Now that #9953 was merged, I rebased this.  Also fixed the mac and kdf commands to properly use ```OPT_MORE_STRING```

**Comment by richsalz** [comment]
ping @mattcaswell who commented over at #10313

File: apps/cms.c [hunk 1]
```diff
@@ -88,9 +88,7 @@ typedef enum OPTION_choice {
 } OPTION_CHOICE;
 
 const OPTIONS cms_options[] = {
-    {OPT_HELP_STR, 1, '-', "Usage: %s [options] cert.pem...\n"},
-    {OPT_HELP_STR, 1, '-',
-        "  cert.pem... recipient certs for encryption\n"},
+    {OPT_HELP_STR, 1, '-', "Usage: %s [options] [cert...]\n"},
```
Comment (t8m) [inline apps/cms.c]: In general to distinguish between literal and replaceable parameter names (as italics cannot be used in usage outputs) the commands use capitalized words for the replaceable texts. But that would be probably a separate PR to do and maybe even not worth it.

File: apps/cms.c [hunk 2]
```diff
@@ -219,6 +217,9 @@ const OPTIONS cms_options[] = {
 
     OPT_R_OPTIONS,
     OPT_V_OPTIONS,
+
+    OPT_PARAMETERS(),
+    {"cert", 0, 0, "Recipient certs, used when encrypting"},
```
Comment (t8m) [inline apps/cms.c]: perhaps `Recipient certs (optional, required and used only when encrypting)` ?

File: apps/verify.c [hunk 3]
```diff
@@ -72,6 +72,9 @@ const OPTIONS verify_options[] = {
     {"sm2-hex-id", OPT_SM2HEXID, 's',
      "Specify a hex ID string to verify an SM2 certificate"},
 #endif
+
+    OPT_PARAMETERS(),
+    {"cert", 0, 0, "Certificate(s) to verify (optional)"},
```
Comment (t8m) [inline apps/verify.c]: Maybe `(optional; stdin used otherwise)` ?

File: apps/cms.c (hunk 1 above)
Comment (richsalz) [reply apps/cms.c]: I'd prefer someone do that as a separate PR if it seems worthwhile.

File: apps/cms.c (hunk 2 above)
Comment (richsalz) [reply apps/cms.c]: I did "Recipient certs (optional; required only when encrypting)"

Comment (richsalz) [reply apps/cms.c]: That wording isn't great either, since you can use -recip.  Sigh. So I made it "used only"

File: apps/verify.c (hunk 3 above)
Comment (richsalz) [reply apps/verify.c]: done.

**Comment by richsalz** [comment]
fixup commit pushed.

File: apps/cms.c (hunk 1 above)
Comment (t8m) [reply apps/cms.c]: That's definitely OK.

File: apps/speed.c [hunk 4]
```diff
@@ -356,6 +356,9 @@ const OPTIONS speed_options[] = {
      "Use specified offset to mis-align buffers"},
 
     OPT_R_OPTIONS,
+
+    OPT_PARAMETERS(),
+    {"algorithm", 0, 0, "Algorithm(s) to test"},
```
[the remark this answers is not in the document]
Comment (t8m) [reply apps/speed.c]: @richsalz What about this one?

[the remark this answers is not in the document]
Comment (richsalz) [reply apps/speed.c]: The behavior if not specified is wonky. It depends on what flags you use. I tried to verify the behavior and got this:
```
; ./apps/openssl speed
Doing md2 for 3s on 16 size blocks: EVP error!
        exit 1
```

But I'll add optional.

[the remark this answers is not in the document]
Comment (t8m) [reply apps/speed.c]: That would be a bug in speed.c. It works fine in 1.1.1.

[the remark this answers is not in the document]
Comment (richsalz) [reply apps/speed.c]: the bug is more fundamental, but I opened #10435 to catch it.  resolving this suyb-thread.

**Comment by t8m** [comment]
Merged to master as 92de469fbd6
